src/libraries/System.Threading.Tasks.Parallel/tests/ParallelForEachAsyncTests.cs

```diff
@@ -117,6 +158,40 @@ static IEnumerable<int> IterateUntilSet(StrongBox<bool> box)
             Assert.InRange(activeWorkers, 0, dop == -1 ? Environment.ProcessorCount : dop);
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [InlineData(-1)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(4)]
+        [InlineData(128)]
+        public async Task Dop_WorkersCreatedRespectingLimitAndTaskScheduler_For(int dop)
+        {
+            bool exit = false;
+            int activeWorkers = 0;
+            var block = new TaskCompletionSource();
+
+            int MaxSchedulerLimit = Math.Min(2, Environment.ProcessorCount);
+
+            Task t = Parallel.ForAsync(long.MinValue, long.MaxValue, new ParallelOptions { MaxDegreeOfParallelism = dop, TaskScheduler = new MaxConcurrencyLevelPassthroughTaskScheduler(MaxSchedulerLimit) }, async (item, cancellationToken) =>
+            {
+                Interlocked.Increment(ref activeWorkers);
+                await block.Task;
+                if (Volatile.Read(ref exit))
+                {
+                    throw new FormatException();
+                }
+            });
+            Assert.False(t.IsCompleted);
+
+            await Task.Delay(20); // give the loop some time to run
```

Comment:
Do you think this delay be a good enough reason to move the test to outerloop? Here and elsewhere.

Reply:
Yes.  Correctness doesn't depend on this delay, it just might not test as much as we want in some circumstances if things take longer than expected.

Reply:
Oh, I misread the question.  No, I don't think this needs to be outerloop.  It's only 20 milliseconds.

Reply:
Oh you're right. I was thinking in seconds.